polygeist/tools/cgeist/Test/Verification/sycl/undeftype.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: clang++ -fsycl -fsycl-device-only -fsycl-targets=spir64-unknown-unknown-syclmlir -O0 -w -emit-mlir -o - %s | FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+template <typename T> SYCL_EXTERNAL void keep(T);
+
+namespace sycl {
+class VecWrapper {
+  float4 v;
+public:
+  SYCL_EXTERNAL explicit VecWrapper(float);
+  SYCL_EXTERNAL float operator[](size_t) const;
+};
+}  // namespace sycl
+
+// CHECK-LABEL:  func.func @_Z16vec_wrapper_testfm(
+// CHECK-SAME:     %[[SPAN:.*]]: f32
+// CHECK-SAME:     %[[INDEX:.*]]: i64
+// CHECK:          sycl.call(%{{.*}}, %[[SPAN]]) {FunctionName = @VecWrapper, MangledFunctionName = @{{.*}}, TypeName = @VecWrapper} : (memref<?x!llvm.struct<(!sycl_vec_f32_4_)>, 4>, f32) -> ()
```

Comment:
We generate a `sycl.call` instead of breaking.

Reply:
I am confused, isn't the sycl namespace reserved for the SYCL runtime? 
IMO end users should not define their type into a namespace reserved for the runtime. This is similar to defining a type in the std namespace, which is reserved for the C++ standard library. 

My opinion is that this example is invalid.